### ISSUE TITLE
[Bug]: Discord rate limit issues

### ISSUE THREAD

**Issue by hotutils** [RogueException/Discord.Net#2512]
### Check The Docs

- [X] I double checked the docs and couldn't find any useful information.

### Verify Issue Source

- [X] I verified the issue was caused by Discord.Net.

### Check your intents

- [X] I double checked that I have the required intents.

### Description

As of yesterday about noon EST, something changed on the discord API side related to this:

https://github.com/discord/discord-api-docs/issues/5646

Many libraries have remained unaffected, but this appears to have affected discord.net framework in some way related to buckets and rate limit detection, possibly related to not respecting "X-RateLimit-Remaining" ??

Our bot is now getting slammed with pre-emptive rate limiting, especially through direct messages. 

I am hoping someone would be able to assist in identifying the root issue and a possible fix.

We are 100% based on slash commands, and simply sending in a few slash commands in a DM with our bot can instantly start reproducing the issues.

### Version

3.8.1

### Logs

2022-11-16 08:32:06.8467|ERROR|HotBot.Bot|Rate Limit: NULL NULL/NULL for "channels/830804335314337862". Reset in 0 seconds, is global false
2022-11-16 08:32:07.1648|ERROR|HotBot.Bot|Rate Limit: "d1240eeffb4c549bd62b579ece0ffd01" 49999/50000 for "channels/830804335314337862/messages". Reset in 259200 seconds, is global false
2022-11-16 08:32:07.3445|ERROR|HotBot.Bot|Rate Limit: NULL NULL/NULL for "channels/993917873221271732". Reset in 0 seconds, is global false
2022-11-16 08:32:07.5698|ERROR|HotBot.Bot|Rate Limit: "d1240eeffb4c549bd62b579ece0ffd01" 49999/50000 for "channels/993917873221271732/messages". Reset in 259200 seconds, is global false
2022-11-16 08:32:07.7607|ERROR|HotBot.Bot|Rate Limit: NULL NULL/NULL for "channels/861358668910166027". Reset in 0 seconds, is global false
2022-11-16 08:32:08.0349|ERROR|HotBot.Bot|Rate Limit: "d1240eeffb4c549bd62b579ece0ffd01" 49999/50000 for "channels/861358668910166027/messages". Reset in 259200 seconds, is global false
2022-11-16 08:32:08.2322|ERROR|HotBot.Bot|Rate Limit: NULL NULL/NULL for "channels/697581888545751110". Reset in 0 seconds, is global false
2022-11-16 08:32:08.6165|ERROR|HotBot.Bot|Rate Limit: "d1240eeffb4c549bd62b579ece0ffd01" 49999/50000 for "channels/697581888545751110/messages". Reset in 259200 seconds, is global false
2022-11-16 08:32:08.8023|ERROR|HotBot.Bot|Rate Limit: NULL NULL/NULL for "channels/605897761576976384". Reset in 0 seconds, is global false
2022-11-16 08:32:09.0539|ERROR|HotBot.Bot|Rate Limit: "d1240eeffb4c549bd62b579ece0ffd01" 49999/50000 for "channels/605897761576976384/messages". Reset in 259200 seconds, is global false

### Packages

Discord.Net.* 3.8.1

**Comment by Baaleos** [comment]
I added the following to a RateLimitCallback to get some debug info: 
Console.WriteLine($"{info.IsGlobal} {info.Limit} {info.Remaining} {info.RetryAfter} {info.Reset} {info.ResetAfter} {info.Bucket} {info.Lag} {info.Endpoint}");

![CB9F5867-6628-4F27-AABC-65AE7C79DAFD](https://user-images.githubusercontent.com/6419940/202260706-caf9545f-c6e4-4188-9b1c-b5482f78cf52.png)

I was on discord.net 3.7.2 when this kicked in yesterday evening, and I have since upgraded to 3.8.1 and still persistent.

the symptom it is causing is that attempting to add more than 2 reactions to a post/message results in the code stalling for 5 minutes. The debug message shows that the rate limit values appear odd- jumping to 0 triggering early rate limit.
